File: node/src/commands/geospatial/GeospatialData.ts
/**
 * Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
 */

/**
 * Represents a geographic position defined by longitude and latitude.
 * The exact limits, as specified by `EPSG:900913 / EPSG:3785 / OSGEO:41001` are the
 * following:
 *
 *   Valid longitudes are from `-180` to `180` degrees.
 *   Valid latitudes are from `-85.05112878` to `85.05112878` degrees.
 */
export class GeospatialData {
    private longitude: number;

    private latitude: number;

    /**
     * Default constructor for GeospatialData.
     *
     * @param longitude - The longitude coordinate.
     * @param latitude - The latitude coordinate.
     */
    constructor(longitude: number, latitude: number) {
        this.longitude = longitude;
        this.latitude = latitude;
    }

    /**
     * Converts GeospatialData into a string[].
     *
     * @returns string[]
     */
    public toArgs(): string[] {
        return [this.longitude.toString(), this.latitude.toString()];
    }
}

File: JNI-java/client/src/main/java/glide/api/models/commands/geospatial/GeospatialData.java
/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
package glide.api.models.commands.geospatial;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * Represents a geographic position defined by longitude and latitude.<br>
 * The exact limits, as specified by <code>EPSG:900913 / EPSG:3785 / OSGEO:41001</code> are the
 * following:
 *
 * <ul>
 *   <li>Valid longitudes are from <code>-180</code> to <code>180</code> degrees.
 *   <li>Valid latitudes are from <code>-85.05112878</code> to <code>85.05112878</code> degrees.
 * </ul>
 */
@Getter
@RequiredArgsConstructor
public final class GeospatialData {
    /** The longitude coordinate. */
    private final double longitude;

    /** The latitude coordinate. */
    private final double latitude;

    /**
     * Converts GeospatialData into a String[].
     *
     * @return String[] An array containing the longtitue and the latitue of the position.
     */
    public String[] toArgs() {
        return new String[] {Double.toString(longitude), Double.toString(latitude)};
    }
}
